seqr/utils/vcf_utils.py

```diff
@@ -0,0 +1,91 @@
+import re
+
+from collections import defaultdict
+
+from seqr.utils.middleware import ErrorsWarningsException
+from seqr.utils.file_utils import file_iter
+
+BLOCK_SIZE = 65536
+
+EXPECTED_META_FIELDS ={
+    'INFO': {
+        'AC': {'Number': 'A', 'Type': 'Integer'},
+        'AN': {'Number': '1', 'Type': 'Integer'},
+        'AF': {'Number': 'A', 'Type': 'Float'}
+    },
+    'FORMAT': {
+        'AD': {'Number': '.', 'Type': 'Integer'},
+        'DP': {'Number': '1', 'Type': 'Integer'},
+        'GQ': {'Number': '1', 'Type': 'Integer'},
+        'GT': {'Number': '1', 'Type': 'String'}
+    }
+}
+
+REQUIRED_HEADERS = ['#CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT']
+
+
+def _validate_vcf_header(header):
+    missing_headers = [h for h in REQUIRED_HEADERS if h not in header ]
+    if missing_headers:
+        missing_fields = ','.join(missing_headers)
+        raise ErrorsWarningsException([f'Required VCF header fields {missing_fields} are missing.'], [])
+
+
+def _validate_vcf_meta(meta):
+    for field, sub_field_meta in EXPECTED_META_FIELDS.items():
+        for sub_field, metadata in sub_field_meta.items():
+            for key, value in metadata.items():
+                if meta.get(field, {}).get(sub_field, {}).get(key) != value:
+                    error = f'VCF header field {field}.{sub_field} and meta information {key}={value} is expected.'
+                    raise ErrorsWarningsException([error], [])
+
+
+def _validate_vcf_data_row(first_line):
```

Comment:
per the ticket and our previous discussion on this PR, we will not be validating both the meta and the first row. Pick one or the other, doing both is redundant, confusing, and hard to maintain

Reply:
Okay.